embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/ApiResponse.kt

```diff
@@ -1,7 +1,21 @@
 package io.embrace.android.embracesdk.comms.api
 
-internal data class ApiResponse<T>(
-    val statusCode: Int?,
-    val headers: Map<String, String>,
-    val body: T?
-)
+/**
+ * ApiResponse is a sealed class that represents the result of an API call.
+ */
+internal sealed class ApiResponse<out T> {
+    /**
+     * Represents a successful API call (status code 200-299)
+     */
+    data class Success<T>(val body: T?, val headers: Map<String, String>?) : ApiResponse<T>()
+
+    /**
+     * Represents a failed API call. (status code 400-499 or 500-599)
+     */
+    data class Failure(val errorMessage: String?, val code: Int, val headers: Map<String, String>?) : ApiResponse<Nothing>()
```

Comment:
Also, any reason why you aren't sub-dividing the different groups of failure so we can handle them without looking at the status code? It would be nice to just have a `when` statement that exhaustively handles all the different classes without us have to look at the status code.